modules/consensus/consensusset.go

```diff
@@ -132,13 +132,28 @@ func New(gateway modules.Gateway, persistDir string) (*ConsensusSet, error) {
 	}
 
 	go func() {
-		// Sync with the network. Don't sync if we are testing because typically we
-		// don't have any mock peers to synchronize with in testing.
-		// TODO: figure out a better way to conditionally do IBD. Otherwise this block will never be tested.
+		// Sync with the network. Don't sync if we are testing because
+		// typically we don't have any mock peers to synchronize with in
+		// testing.
 		if build.Release != "testing" {
-			cs.threadedInitialBlockchainDownload()
+			// A blocking operation which will grab a thread group. We are in a
+			// virgin goroutine right now, so calling the threaded function
+			// without a goroutine is okay.
+			err = cs.threadedInitialBlockchainDownload()
+			if err != nil {
+				return
+			}
 		}
 
+		// threadedInitialBlockchainDownload will release the threadgroup 'Add'
+		// it was holding, so another needs to be grabbed to finish off this
+		// goroutine.
```

Comment:
Shouldn't be. If Shutdown is called early, the RPCs will never be registered and then never cleaned up. If Flush is called, the RPCs may not be registered upon completion.

If you want control over waiting until the RPCs are registered, you'll need a separate thread group (or maybe just a WaitGroup). IBD can be a very long process, it does not make sense to block shutdown until IBD has completed.

Maybe it does, if you add in some intelligent clean-shutdown code.